kani-compiler/src/kani_middle/reachability.rs

```diff
@@ -494,6 +512,12 @@ impl<'a, 'tcx> MirVisitor<'tcx> for MonoItemsFnCollector<'a, 'tcx> {
                                 // implement the same traits as those in the
                                 // original function/method. A trait mismatch shows
                                 // up here, when we try to resolve a trait method
+
+                                // FIXME: This assumes the type resolving the
+                                // trait is the first argument, but that isn't
+                                // necessarily true. It could be any argument or
+                                // even the return type, for instance for a
+                                // trait like `FromIterator`.
```

Comment:
Do we have an GitHub issue related to this? Any reason to not have one here?

Reply:
No. This is also not my code. I was just wondering how it could dispatch on the first argument and how that would be correct so I crafted an example that would break it and added the comment so if it comes up whoever fixes it has something to go on.